File: docs/pa/files/pa1.py
# PA1 (fall 2026) Python skeleton code
#
# This file shows how to read and parse the input, which is provided via
# standard input


edges = [] # will be a list of 4-tuples: source node (string), 
		   # dest node (string), available capacity, total capacity
nodes = [] # will be the string names of the nodes
start_node = end_node = None


def read_input():
	global edges, nodes, start_node, end_node
	parts = input().strip().split(" ")
	num_edges = int(parts[0])
	start_node, end_node = parts[1], parts[2]
	nodeset = set()
	for _ in range(num_edges):
		parts = input().split(" ")
		nodeset.add(parts[0])
		nodeset.add(parts[1])
		parts[2] = int(parts[2])
		parts[3] = int(parts[3])
		edges.append( tuple(parts) )
	nodes = list(nodeset)


# read the input from standard input
read_input()


# REMOVE THESE LINES BEFORE SUBMISSION
# any extra output will cause the program to be marked incorrect
print(f"Start node is {start_node}, end node is {end_node}")
print(f"The list of {len(nodes)} nodes:",nodes)
print(f"The list of {len(edges)} edges:",edges)

File: docs/pa/files/PA1.java
// PA1 (fall 2026) Python skeleton code
//
// This file shows how to read and parse the input, which is provided via
// standard input

import java.util.*;

class Edge {
	public int available_capacity, total_capacity;
	public String start_node, end_node;

	Edge(Scanner stdin) {
		start_node = stdin.next();
		end_node = stdin.next();
		available_capacity = stdin.nextInt();
		total_capacity = stdin.nextInt();
	}

	public String toString() {
		return "('" + start_node + "', '" + end_node + "', " + available_capacity +
			", " + total_capacity + ")";
	}
}

public class PA1 {
	static ArrayList<String> nodes = new ArrayList<String>();
	static ArrayList<Edge> edges = new ArrayList<Edge>();
	static String start_node, end_node;

	public static void main (String[] args) {
		Scanner stdin = new Scanner(System.in);
		Set<String> nodeset = new HashSet<String>();
		int num_edges = stdin.nextInt();
		start_node = stdin.next();
		end_node = stdin.next();
		for ( int i = 0; i < num_edges; i++ ) {
			Edge e = new Edge(stdin);
			nodeset.add(e.start_node);
			nodeset.add(e.end_node);
			edges.add(e);
		}
		nodes.addAll(nodeset);

		// REMOVE THESE LINES BEFORE SUBMISSION
		// any extra output will cause the program to be marked incorrect
		System.out.println("Start node is " + start_node + ", end node is " + end_node);
		System.out.println("The list of " + nodes.size() + " nodes: " + nodes);
		System.out.println("The list of " + edges.size() + " edges: " + edges);
	}
}
